services/SchedulableTransactionBody.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "SystemDelete.proto";
+import "SystemUndelete.proto";
+import "Freeze.proto";
+
+import "ContractCall.proto";
+import "ContractCreate.proto";
+import "ContractUpdate.proto";
+
+import "CryptoCreate.proto";
+import "CryptoDelete.proto";
+import "CryptoTransfer.proto";
+import "CryptoUpdate.proto";
+
+import "FileAppend.proto";
+import "FileCreate.proto";
+import "FileDelete.proto";
+import "FileUpdate.proto";
+
+import "Duration.proto";
+import "BasicTypes.proto";
+import "ContractDelete.proto";
+
+import "ConsensusCreateTopic.proto";
+import "ConsensusUpdateTopic.proto";
+import "ConsensusDeleteTopic.proto";
+import "ConsensusSubmitMessage.proto";
+
+import "TokenCreate.proto";
+import "TokenFreezeAccount.proto";
+import "TokenUnfreezeAccount.proto";
+import "TokenGrantKyc.proto";
+import "TokenRevokeKyc.proto";
+import "TokenDelete.proto";
+import "TokenUpdate.proto";
+import "TokenMint.proto";
+import "TokenBurn.proto";
+import "TokenWipeAccount.proto";
+import "TokenAssociate.proto";
+import "TokenDissociate.proto";
+
+import "ScheduleDelete.proto";
+
+/* A schedulable transaction. Note that the global/dynamic system property 
+<tt>scheduling.whitelist</tt> controls which transaction types may be scheduled. 
+In Hedera Services 0.13.0, it will include only <tt>CryptoTransfer</tt> and
+<tt>ConsensusSubmitMessage</tt> functions. */
+message SchedulableTransactionBody {
+  uint64 transactionFee = 1; // The maximum transaction fee the client is willing to pay
```

Comment:
When the transaction is executed will you be constructing an empty `TransactionBody` then copy all these fields over explicitly? Or will you be reconstituting a `TransactionBody` from the `SchedulableTransactionBody` bytes e.g. `TransactionBody.parseFrom(scheduleableTransactionBodyBytes)`. If the latter, you will need to adjust these field indexes to match `TransactionBody`.

Reply:
I've done [the former](https://github.com/hashgraph/hedera-services/blob/2cae66315472f6c844d532b59b0201d195dcb697/hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleSchedule.java#L377) in my Services PR. 👍